mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java

```diff
@@ -373,4 +745,84 @@ private Mailet initMailet() throws MessagingException {
         return mailet;
     }
 
+    @Test
+    public void fileNameMatchesShouldReturnFalseWhenPatternDoesntMatch() throws Exception {
+        StripAttachment mailet = new StripAttachment();
+        FakeMailetConfig mci = FakeMailetConfig.builder()
+                .setProperty("pattern", ".*pattern.*")
+                .build();
+        mailet.init(mci);
+        
+        assertThat(mailet.fileNameMatches("not matching")).isFalse();
+    }
+
+    @Test
+    public void fileNameMatchesShouldReturnTrueWhenPatternMatches() throws Exception {
+        StripAttachment mailet = new StripAttachment();
+        FakeMailetConfig mci = FakeMailetConfig.builder()
+                .setProperty("pattern", ".*pattern.*")
+                .build();
+        mailet.init(mci);
+        
+        assertThat(mailet.fileNameMatches("I've got a pattern.")).isTrue();
+    }
+
+    @Test
+    public void fileNameMatchesShouldReturnFalseWhenNotPatternMatches() throws Exception {
+        StripAttachment mailet = new StripAttachment();
+        FakeMailetConfig mci = FakeMailetConfig.builder()
+                .setProperty("notpattern", ".*pattern.*")
+                .build();
+        mailet.init(mci);
+        
+        assertThat(mailet.fileNameMatches("I've got a pattern.")).isFalse();
+    }
+
+    @Test
+    public void fileNameMatchesShouldReturnTrueWhenNotPatternDoesntMatch() throws Exception {
+        StripAttachment mailet = new StripAttachment();
+        FakeMailetConfig mci = FakeMailetConfig.builder()
+                .setProperty("notpattern", ".*pattern.*")
+                .build();
+        mailet.init(mci);
+        
+        assertThat(mailet.fileNameMatches("not matching")).isTrue();
+    }
+
+    @Test
+    public void fileNameMatchesShouldReturnFalseWhenPatternAndNotPatternAreTheSame() throws Exception {
+        StripAttachment mailet = new StripAttachment();
+        FakeMailetConfig mci = FakeMailetConfig.builder()
+                .setProperty("pattern", ".*pattern.*")
+                .setProperty("notpattern", ".*pattern.*")
+                .build();
+        mailet.init(mci);
+        
+        assertThat(mailet.fileNameMatches("not matching")).isFalse();
+        assertThat(mailet.fileNameMatches("I've got a pattern.")).isFalse();
+    }
+
+    @Test
+    public void fileNameMatchesShouldReturnTrueWhenPatternMatchesAndNotPatternDoesntMatch() throws Exception {
+        StripAttachment mailet = new StripAttachment();
+        FakeMailetConfig mci = FakeMailetConfig.builder()
+                .setProperty("pattern", ".*pattern.*")
+                .setProperty("notpattern", ".*notpattern.*")
+                .build();
+        mailet.init(mci);
+        
+        assertThat(mailet.fileNameMatches("I've got a pattern.")).isTrue();
```

Comment:
If you show me thi method, then you should also do the usual edge cases (here null)